ape_safe/client.py

```diff
@@ -359,7 +359,7 @@ def post_signature(
 class MockSafeClient(BaseSafeClient, ManagerAccessMixin):
     def __init__(self, contract: ContractInstance):
         self.contract = contract
-        self.transactions: Dict[SafeTxID, SafeApiTxData] = {}
+        self.transactions: Dict[SafeTx | SafeTxID, SafeApiTxData] = {}
```

Comment:
this python 3.10 syntax (the `|` union). You need to use the `Union` type to support < python 3.10